state/aws/dynamodb/dynamodb.go

```diff
@@ -227,8 +233,8 @@ func (d *StateStore) BulkSet(ctx context.Context, req []state.SetRequest) error
 func (d *StateStore) Delete(ctx context.Context, req *state.DeleteRequest) error {
 	input := &dynamodb.DeleteItemInput{
 		Key: map[string]*dynamodb.AttributeValue{
-			"key": {
-				S: aws.String(populatePartitionMetadata(req.Key, req.Metadata)),
+			d.partitionKey: {
```

Comment:
In the current version, the partition key is read from the request's metadata. Now, it's read from the component's metadata.

Shouldn't we continue allowing passing a partition key for each request?

Reply:
Since the DynamoDB table needs to be created before hand and once the table is created the partition key can not be changed through the request, I felt that specifying the partition key only through the component's metadata made more sense. Unless, I'm missing something.

Reply:
Oh I see. That's ok then.